components/director/internal/open_resource_discovery/client.go

```diff
@@ -69,25 +84,48 @@ func (c *client) FetchOpenResourceDiscoveryDocuments(ctx context.Context, app *m
 	}
 
 	docs := make([]*Document, 0)
+	mutex := sync.RWMutex{}
+	wg := sync.WaitGroup{}
+	queue := make(chan DocumentDetails)
+	var fetchDocError error = nil
+
+	for i := 0; i < c.config.maxParallelDocuments; i++ {
+		wg.Add(1)
+		go c.fetchDocumentsContent(ctx, &fetchDocError, queue, &wg, &mutex, &docs, baseURL, tenantValue)
+	}
+
 	for _, docDetails := range config.OpenResourceDiscoveryV1.Documents {
+		queue <- docDetails
+	}
+
+	close(queue)
+	wg.Wait()
+
+	return docs, baseURL, fetchDocError
+}
+
+func (c *client) fetchDocumentsContent(ctx context.Context, fetchDocError *error, queue chan DocumentDetails, wg *sync.WaitGroup, mutex *sync.RWMutex, docs *[]*Document, baseURL, tenantValue string) {
+	defer wg.Done()
+	for docDetails := range queue {
```

Comment:
Double check that this range loop is exited after the `queue` channel is closed on line 101